pdoc/extract.py

```diff
@@ -46,6 +47,13 @@ def walk_specs(specs: Sequence[Union[Path, str]]) -> dict[str, None]:
     """
     all_modules: dict[str, None] = {}
     for spec in specs:
+
+        str_spec = str(spec)
+        if len(str_spec) > 0 and str_spec[0] == '!':
```

Comment:
Nit: Let's do ` if isinstance(spec, str) and spec.startswith("!")`, that's probably a bit more Pythonic. 😃

Reply:
Please see my comment below for this part

Reply:
Fixed